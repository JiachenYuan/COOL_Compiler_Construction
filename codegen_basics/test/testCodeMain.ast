#13
_program
  #13
  _class
    Main
    Object
    "testCodeMain.cl"
    (
    #12
    _method
      main
      Int
      #11
      _block
        #8
        _let
          b
          Bool
          #5
          _bool
            1
          : Bool
          #8
          _block
            #6
            _object
              b
            : Bool
            #7
            _assign
              b
              #7
              _bool
                0
              : Bool
            : Bool
          : Bool
        : Bool
        #10
        _int
          123
        : Int
      : Int
    )
