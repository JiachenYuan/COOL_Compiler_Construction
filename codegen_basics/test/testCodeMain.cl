class Main {
    main(): Int {
        {

          let b: Bool <- true in {
            b;
            b <- false;
          };

          123;
        }
    };
};