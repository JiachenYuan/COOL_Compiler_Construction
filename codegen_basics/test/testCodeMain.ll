; ModuleID = 'module'
source_filename = "module"

@.str = internal constant [25 x i8] c"Main.main() returned %d\0A\00"

declare i32 @strcmp(ptr, ptr)

declare i32 @printf(ptr, ...)

declare void @abort()

declare ptr @malloc(i32)

define i32 @Main_main() {
entry:
  %0 = alloca i1, align 1
  store i1 true, ptr %0, align 1
  %1 = load i1, ptr %0, align 1
  store i1 false, ptr %0, align 1
  ret i32 123

abort:                                            ; No predecessors!
  call void @abort()
  unreachable
}

define i32 @main() {
entry:
  %0 = call i32 @Main_main()
  %1 = call i32 (ptr, ...) @printf(ptr @.str, i32 %0)
  ret i32 0
}
