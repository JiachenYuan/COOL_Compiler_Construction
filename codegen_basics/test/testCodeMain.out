Main.main() returned 123
